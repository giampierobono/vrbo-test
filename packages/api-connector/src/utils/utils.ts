export const getRandomNumber = (minValue: number, maxValue: number) =>
  Math.round(
    parseFloat(
      Math.min(
        minValue + Math.random() * (maxValue - minValue),
        maxValue,
      ).toFixed(2),
    ) * 2,
  ) / 2;
