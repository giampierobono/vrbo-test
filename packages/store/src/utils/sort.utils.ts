import { PropertiesSortByConfig } from "@vrbo/data-models";
import { ifElse, negate, prop, sortBy } from "ramda";
import { pipe } from "rxjs";

export const sortByConfig = (sortConfig: PropertiesSortByConfig) =>
  ifElse(
    () => sortConfig.asc,
    sortBy(prop(sortConfig.prop)),
    sortBy(pipe(prop(sortConfig.prop), negate))
  );
