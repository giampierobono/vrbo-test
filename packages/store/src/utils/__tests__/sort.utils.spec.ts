import { sortByConfig } from "../sort.utils";

describe("Sort utils", () => {
  const arrayToSort = [
    {
      propString: "R",
      propNum: 3
    },
    {
      propString: "E",
      propNum: 13
    },
    {
      propString: "D",
      propNum: 34
    },
    {
      propString: "A",
      propNum: 54
    },
    {
      propString: "H",
      propNum: 0
    },
    {
      propString: "F",
      propNum: 24
    }
  ];

  describe("sort by prop string", () => {
    it("should sort ascending", () => {
      expect(
        sortByConfig({ prop: "propString", asc: true } as any)(arrayToSort)
      ).toMatchSnapshot();
    });

    it("should sort descending", () => {
      expect(
        sortByConfig({ prop: "propString", asc: false } as any)(arrayToSort)
      ).toMatchSnapshot();
    });
  });

  describe("sort by prop num", () => {
    it("should sort ascending", () => {
      expect(
        sortByConfig({ prop: "propNum", asc: true } as any)(arrayToSort)
      ).toMatchSnapshot();
    });

    it("should sort descending", () => {
      expect(
        sortByConfig({ prop: "propNum", asc: false } as any)(arrayToSort)
      ).toMatchSnapshot();
    });
  });
});
