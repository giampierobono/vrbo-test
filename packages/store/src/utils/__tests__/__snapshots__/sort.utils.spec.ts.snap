// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort utils sort by prop num should sort ascending 1`] = `
Array [
  Object {
    "propNum": 0,
    "propString": "H",
  },
  Object {
    "propNum": 3,
    "propString": "R",
  },
  Object {
    "propNum": 13,
    "propString": "E",
  },
  Object {
    "propNum": 24,
    "propString": "F",
  },
  Object {
    "propNum": 34,
    "propString": "D",
  },
  Object {
    "propNum": 54,
    "propString": "A",
  },
]
`;

exports[`Sort utils sort by prop num should sort descending 1`] = `
Array [
  Object {
    "propNum": 54,
    "propString": "A",
  },
  Object {
    "propNum": 34,
    "propString": "D",
  },
  Object {
    "propNum": 24,
    "propString": "F",
  },
  Object {
    "propNum": 13,
    "propString": "E",
  },
  Object {
    "propNum": 3,
    "propString": "R",
  },
  Object {
    "propNum": 0,
    "propString": "H",
  },
]
`;

exports[`Sort utils sort by prop string should sort ascending 1`] = `
Array [
  Object {
    "propNum": 54,
    "propString": "A",
  },
  Object {
    "propNum": 34,
    "propString": "D",
  },
  Object {
    "propNum": 13,
    "propString": "E",
  },
  Object {
    "propNum": 24,
    "propString": "F",
  },
  Object {
    "propNum": 0,
    "propString": "H",
  },
  Object {
    "propNum": 3,
    "propString": "R",
  },
]
`;

exports[`Sort utils sort by prop string should sort descending 1`] = `
Array [
  Object {
    "propNum": 3,
    "propString": "R",
  },
  Object {
    "propNum": 13,
    "propString": "E",
  },
  Object {
    "propNum": 34,
    "propString": "D",
  },
  Object {
    "propNum": 54,
    "propString": "A",
  },
  Object {
    "propNum": 0,
    "propString": "H",
  },
  Object {
    "propNum": 24,
    "propString": "F",
  },
]
`;
